### DISCUSSION TITLE
Get data in router URL

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#14329] [Help] [answered]
Hi, I use  `router.pathname` and i will get this `/item/[itemId]` 
but how can I get [itemId] value ? 
Yep, I can use window.localtion.href  but I prefer use something of nextjs
`

**Comment by ijjk** [comment, marked as answer]
Hi, the `itemId` value is available under `router.query`, see more information related to this in our docs here https://nextjs.org/docs/routing/dynamic-routes

**Reply by IRediTOTO** [reply]
Thank you so much :| I got it, I tried to run router.pathname in an useEffect and it will be undefined
